samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.c

```diff
@@ -961,21 +991,22 @@ static void gen_onpowerup_status(struct bt_mesh_model *model,
 
 /* Generic Power OnOff Setup Server message handlers */
 
-static void gen_onpowerup_publisher(struct bt_mesh_model *model)
+static void gen_onpowerup_publish(struct bt_mesh_model *model)
 {
+	int err;
 	struct net_buf_simple *msg = model->pub->msg;
 	struct generic_onpowerup_state *state = model->user_data;
 
-	if (model->pub->addr != BT_MESH_ADDR_UNASSIGNED) {
-		int err;
+	if (model->pub->addr == BT_MESH_ADDR_UNASSIGNED) {
+		return;
+	}
 
-		bt_mesh_model_msg_init(msg, BT_MESH_MODEL_OP_2(0x82, 0x12));
-		net_buf_simple_add_u8(msg, state->onpowerup);
+	bt_mesh_model_msg_init(msg, BT_MESH_MODEL_OP_2(0x82, 0x12));
```

Comment:
Why do you have these magic 0x82, 0x12 values here when in other places you have proper defines for the opcodes (e.g. `BT_MESH_MODEL_OP_GEN_LEVEL_STATUS`). Please use proper descriptive defines instead of magic values. Otherwise you're just making it hard to follow the code and understand exactly which message is being sent.

Reply:
I will take care of this in separate PR.

Reply:
Done !!